pkg/apis/beat/v1beta1/beat_types.go

```diff
@@ -99,6 +106,29 @@ type DeploymentSpec struct {
 	Strategy appsv1.DeploymentStrategy `json:"strategy,omitempty"`
 }
 
+type Monitoring struct {
+	// Metrics holds references to Elasticsearch clusters which receive monitoring data from this Beats resource.
+	// +kubebuilder:validation:Optional
+	Metrics MetricsMonitoring `json:"metrics,omitempty"`
+	// Logs holds references to Elasticsearch clusters which receive log data from this Beats resource.
+	// +kubebuilder:validation:Optional
+	Logs LogsMonitoring `json:"logs,omitempty"`
+}
+
+type MetricsMonitoring struct {
+	// ElasticsearchRefs is a reference to a list of monitoring Elasticsearch clusters running in the same Kubernetes cluster.
+	// Due to existing limitations, only a single Elasticsearch cluster is currently supported.
+	// +kubebuilder:validation:Required
+	ElasticsearchRefs []commonv1.ObjectSelector `json:"elasticsearchRefs,omitempty"`
+}
+
+type LogsMonitoring struct {
+	// ElasticsearchRefs is a reference to a list of monitoring Elasticsearch clusters running in the same Kubernetes cluster.
+	// Due to existing limitations, only a single Elasticsearch cluster is currently supported.
+	// +kubebuilder:validation:Required
+	ElasticsearchRefs []commonv1.ObjectSelector `json:"elasticsearchRefs,omitempty"`
+}
```

Comment:
I am thinking these types are probably identical between Elasticsearch, Kibana and Beats and could be shared? But maybe there was a reason not to as the original implementation chose to redefine them for each application?

Reply:
@pebrc I have updated this PR to move all of these *Monitoring objects to v1.common package, but this does change the generated docs to be less specific.  Lmk what you think.